infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategyTest.java

```diff
@@ -324,6 +348,127 @@ public void shouldDeletePVCsIfPersistAttributeIsSetToTrueInWorkspaceConfigWhenCl
+    Page workspaces = mock(Page.class);
+    KubernetesPersistentVolumeClaims persistentVolumeClaims =
+        mock(KubernetesPersistentVolumeClaims.class);
+
+    when(workspaceManager.getWorkspaces(anyString(), eq(false), anyInt(), anyLong()))
+        .thenReturn((workspaces));
+    when(workspaces.isEmpty()).thenReturn(true);
+
+    WorkspaceConfig workspaceConfig = mock(WorkspaceConfig.class);
+    when(workspace.getConfig()).thenReturn(workspaceConfig);
+
+    Map<String, String> workspaceConfigAttributes = new HashMap<>();
+    when(workspaceConfig.getAttributes()).thenReturn(workspaceConfigAttributes);
+    workspaceConfigAttributes.put(PERSIST_VOLUMES_ATTRIBUTE, "true");
+
+    KubernetesNamespace ns = mock(KubernetesNamespace.class);
+    when(factory.get(eq(workspace))).thenReturn(ns);
+    when(ns.persistentVolumeClaims()).thenReturn(persistentVolumeClaims);
+
+    EnvironmentContext context = new EnvironmentContext();
+    context.setSubject(new SubjectImpl("user", "id123", "token123", false));
+    EnvironmentContext.setCurrent(context);
+
+    // when
+    commonPVCStrategy.cleanup(workspace);
+
+    // then
+    verify(ns).persistentVolumeClaims();
+    verify(persistentVolumeClaims).delete(PVC_NAME);
+    verify(pvcSubPathHelper, never()).removeDirsAsync(WORKSPACE_ID, "ns", PVC_NAME, WORKSPACE_ID);
+  }
+
+  @Test
+  public void shouldNotDeleteCommonPVCIfUserHasWorkspaces() throws Exception {
+    // given
+    Workspace workspace = mock(Workspace.class);
+    Page workspaces = mock(Page.class);
+    KubernetesPersistentVolumeClaims persistentVolumeClaims =
+        mock(KubernetesPersistentVolumeClaims.class);
+
+
+    when(workspaceManager.getWorkspaces(anyString(), eq(false), anyInt(), anyLong()))
+        .thenReturn((workspaces));
+    when(workspaces.isEmpty()).thenReturn(false);
+    when(workspace.getId()).thenReturn(WORKSPACE_ID);
+
+    WorkspaceConfig workspaceConfig = mock(WorkspaceConfig.class);
+    when(workspace.getConfig()).thenReturn(workspaceConfig);
+
+    Map<String, String> workspaceConfigAttributes = new HashMap<>();
+    when(workspaceConfig.getAttributes()).thenReturn(workspaceConfigAttributes);
+    workspaceConfigAttributes.put(PERSIST_VOLUMES_ATTRIBUTE, "true");
+
+    KubernetesNamespace ns = mock(KubernetesNamespace.class);
+    when(factory.get(eq(workspace))).thenReturn(ns);
+    when(ns.getName()).thenReturn("ns");
+
+    EnvironmentContext context = new EnvironmentContext();
+    context.setSubject(new SubjectImpl("user", "id123", "token123", false));
+    EnvironmentContext.setCurrent(context);
+
+    // when
+    commonPVCStrategy.cleanup(workspace);
+
+    // then
+    verify(ns, never()).persistentVolumeClaims();
+    verify(persistentVolumeClaims, never()).delete(PVC_NAME);
+    verify(pvcSubPathHelper).removeDirsAsync(WORKSPACE_ID, "ns", PVC_NAME, WORKSPACE_ID);
+  }
+
+  @Test
+  public void shouldNotCheckTotalNumberOfWorkspacesIfDefaultNamespaceContainsPlaceholder()
```

Comment:
What is this testing? It doesn't seem related to the changes as far as I can tell...

Reply:
should be removed since the check was related to https://github.com/eclipse-che/che-server/pull/16#discussion_r641895137

Reply:
thank you, fixed